packages/migrate/src/commands/DbPull.ts

```diff
@@ -144,22 +146,33 @@ Set composite types introspection depth to 2 levels
       throw new NoSchemaFoundError()
     }
 
-    let schema: string | null = null
+    const schema = await match({ url, schemaPath })
+      .when(
+        (input): input is { url: string | undefined; schemaPath: string } => input.schemaPath !== null,
+        async (input) => {
+          const rawSchema = fs.readFileSync(input.schemaPath, 'utf-8')
 
-    // Makes sure we have a schema to pass to the engine
-    if (url) {
-      if (schemaPath) {
-        schema = this.urlToDatasource(url)
-        const rawSchema = fs.readFileSync(schemaPath, 'utf-8')
-        schema += removeDatasource(rawSchema)
-      } else {
-        schema = this.urlToDatasource(url)
-      }
-    } else if (schemaPath) {
-      schema = fs.readFileSync(schemaPath, 'utf-8')
-    } else {
-      throw new Error('Could not find a `schema.prisma` file')
-    }
+          if (input.url) {
+            const config = await getConfig({
+              datamodel: rawSchema,
+              ignoreEnvVarErrors: true,
+            })
+            const provider = config.datasources[0]?.provider
+            const schema = this.urlToDatasource(input.url, provider) + removeDatasource(rawSchema)
```

Comment:
```suggestion
            const schema = `${this.urlToDatasource(input.url, provider)}${removeDatasource(rawSchema)}`
```
Usually, it's a safer default to go for template string concatenation
It should do the same in the end.

Reply:
Right, check the new commit!